mono/metadata/appdomain.c

```diff
@@ -706,56 +706,102 @@ mono_domain_has_type_resolve (MonoDomain *domain)
 
 /**
  * mono_domain_try_type_resolve:
- * \param domain application domainwhere the name where the type is going to be resolved
+ * \param domain application domain in which to resolve the type
  * \param name the name of the type to resolve or NULL.
  * \param tb A \c System.Reflection.Emit.TypeBuilder, used if name is NULL.
  *
  * This routine invokes the internal \c System.AppDomain.DoTypeResolve and returns
- * the assembly that matches name.
- *
- * If \p name is null, the value of \c ((TypeBuilder)tb).FullName is used instead
+ * the assembly that matches name, or ((TypeBuilder)tb).FullName.
  *
  * \returns A \c MonoReflectionAssembly or NULL if not found
  */
 MonoReflectionAssembly *
 mono_domain_try_type_resolve (MonoDomain *domain, char *name, MonoObject *tb)
 {
 	MonoError error;
-	MonoReflectionAssembly *ret = mono_domain_try_type_resolve_checked (domain, name, tb, &error);
+	g_assert (domain);
+	g_assert (name || tb);
```

Comment:
This doesn't give enough information in case of crash, as you wouldn't know if it's `name` or `tb` which is `NULL`.

Reply:
When it fails, they are both null.